### ISSUE TITLE
Non-terminal wildcards in select_atoms should raise an error

### ISSUE THREAD

**Issue by lilyminium** [MDAnalysis/mdanalysis#2370]
**Expected behavior**

Currently, only terminal/stemming wildcards (*) are supported by `u.select_atoms`. Putting the asterisk anywhere else should raise an error or warning of some kind.

**Actual behavior**

Instead, the code ignores anything after the * and fails silently.

**Code to reproduce the behavior**

```python
>>> import MDAnalysis as mda
>>> mda.__version__
'0.20.1'
>>> from MDAnalysis.tests.datafiles import *
>>> psf = mda.Universe(PSF)
>>> psf.select_atoms("resname *E").residues.resnames[:10]
array(['MET', 'ARG', 'ILE', 'ILE', 'LEU', 'LEU', 'GLY', 'ALA', 'PRO',
       'GLY'], dtype=object)
>>> psf.select_atoms("resname G*Y").residues.resnames[:10]
array(['GLY', 'GLY', 'GLY', 'GLY', 'GLN', 'GLN', 'GLU', 'GLY', 'GLN',
       'GLY'], dtype=object)
```

**Currently version of MDAnalysis**

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 0.20.1
- Which version of Python (`python -V`)? 3.6 or 3.7
- Which operating system? MacOS Mojave
